rust/src/lockfile.rs

```diff
@@ -213,7 +214,7 @@ mod ffi {
                 let map = lockfile.packages
                     .into_iter()
                     .fold(HashMap::<String, String>::new(), |mut acc, (k, v)| {
-                        acc.insert(format!("{}-{}", k, v.evra), v.digest);
+                        acc.insert(format!("{}-{}", k, v.evra), v.digest.unwrap_or("".into()));
```

Comment:
Yeah, this is on purpose. Ideally, we'd shove a `NULL` there in the final `GHashTable` instead. But right now, we're relying on glib's `HashMap` to `GHashTable` translator, which doesn't make  this straightforward. (But ideally ideally, we'd convert the code that checks those digests to Rust! :) )

We could build the `GHashTable` ourselves (I played with this some more before writing this), but honestly I don't think it's worth it. E.g. you have to take care to use the right allocator, and converting strings all over the place. It would feel like trying to write C in Rust. So when possible, unless it's to avoid really costly overhead, I think we should try to rely on glib's translators?

The code on the C side checks for `NULL` as well as an empty string.